File: Examples/VisualBasic/Thunderbird/GetMarkerInformation.vb
Imports System.IO
Imports Aspose.Email.Formats.Mbox
Imports Aspose.Email.Mail

Namespace Aspose.Email.Examples.VisualBasic.Email.Thunderbird
    Public Class GetMarkerInformation
        Public Shared Sub Run()
            Using stream As New FileStream("inbox.dat", FileMode.Open, FileAccess.Read)
                Using reader As New MboxrdStorageReader(stream, False)
                    Dim fromMarker As String = Nothing
                    Dim msg As MailMessage = reader.ReadNextMessage(fromMarker)
                    'While (InlineAssignHelper(msg, reader.ReadNextMessage(fromMarker))) IsNot Nothing
                    Do While Not msg Is Nothing
                        Console.WriteLine(fromMarker)
                        msg.Dispose()
                        msg = reader.ReadNextMessage(fromMarker)
                    Loop
                End Using
            End Using
            Using writeStream As New FileStream("inbox.dat", FileMode.Create, FileAccess.Write)
                Using writer As New MboxrdStorageWriter(writeStream, False)
                    Dim fromMarker As String = Nothing
                    Dim msg As MailMessage = MailMessage.Load("input.eml")
                    writer.WriteMessage(msg, fromMarker)

                    Console.WriteLine(fromMarker)
                End Using
            End Using
            'ExEnd: GetMarkerInformation
        End Sub
    End Class
End Namespace

File: Examples/Aspose.Email.Examples/MBOX/GetMarkerInformation.cs
using Aspose.Email.Storage.Mbox;
using System;
using System.IO;

namespace Aspose.Email.Examples.MBOX
{
    class GetMarkerInformation
    {
        public static void Run()
        {

            using (FileStream stream = new FileStream(Data.Mbox + "ExampleMbox.mbox", FileMode.Open, FileAccess.Read))
            using (MboxrdStorageReader reader = new MboxrdStorageReader(stream, new MboxLoadOptions()))
            {
                MailMessage msg;
                string fromMarker = null;
                while ((msg = reader.ReadNextMessage(out fromMarker)) != null)
                {
                    Console.WriteLine(fromMarker);

                    msg.Dispose();
                }
            }

            using (FileStream writeStream = new FileStream(Data.Out + "ExampleMbox.mbox", FileMode.Create, FileAccess.Write))
            using (MboxrdStorageWriter writer = new MboxrdStorageWriter(writeStream, new MboxSaveOptions()))
            {
                string fromMarker = null;
                MailMessage msg = MailMessage.Load(Data.Mbox + "EmailWithAttandEmbedded.eml");
                writer.WriteMessage(msg, out fromMarker);

                Console.WriteLine(fromMarker);
            }
        }
    }
}
